pkg/certificate/reconciler_test.go

```diff
@@ -0,0 +1,216 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	tassert "github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/compute/kube"
+	"github.com/openservicemesh/osm/pkg/constants"
+	fakeConfigClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned/fake"
+	"github.com/openservicemesh/osm/pkg/k8s"
+	"github.com/openservicemesh/osm/pkg/k8s/informers"
+	"github.com/openservicemesh/osm/pkg/messaging"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func TestCheckAndUpdate(t *testing.T) {
+	a := tassert.New(t)
+
+	mrcClient := &fakeMRCClient{}
+	checkStatus := func(ctx context.Context, mrc *v1alpha2.MeshRootCertificate) (bool, error) {
+		if mrc.Status.State != constants.MRCStatePending {
```

Comment:
I don't know how I feel about depending on the top-level `status.state`. The certificateStatuses are what should be informing the condition updates